pkg/cloud/aws/services/ec2/gateways.go

```diff
@@ -121,12 +127,37 @@ func (s *Service) createInternetGateway() (*ec2.InternetGateway, error) {
 	}
 
 	s.scope.Info("Created internet gateway for VPC", "vpc-id", s.scope.VPC().ID)
-	_, err = s.scope.EC2.AttachInternetGateway(&ec2.AttachInternetGatewayInput{
+
+	tagParams := s.getGatewayTagParams(*ig.InternetGateway.InternetGatewayId)
+	tagApply := &tags.ApplyParams{
+		EC2Client:   s.scope.EC2,
+		BuildParams: tagParams,
+	}
+	tag := func() (bool, error) {
+		if err := tags.Apply(tagApply); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	if err := wait.WaitForWithRetryable(wait.NewBackoff(), tag, awserrors.InternetGatewayNotFound); err != nil {
```

Comment:
Yeah, while strictly not required, since the tags would still be applied after the create call finishes, but this brings it in line with the other "services".